cmd/config-manager/main.go

```diff
@@ -29,33 +29,66 @@ import (
 )
 
 const (
-	tomlFilePath = "/etc/containerd/config.toml"
-	nriPluginKey = "io.containerd.nri.v1.nri"
-	disableKey   = "disable"
-	replaceMode  = "replace"
-	resultDone   = "done"
-	unit         = "containerd.service"
+	containerdConfigFile = "/etc/containerd/config.toml"
+	crioConfigFile       = "/etc/crio/crio.conf.d/10-enable-nri.conf"
+	nriPluginKey         = "io.containerd.nri.v1.nri"
+	replaceMode          = "replace"
+	resultDone           = "done"
+	containerdUnit       = "containerd.service"
+	crioUnit             = "crio.service"
 )
 
 func main() {
-	tomlMap, err := readConfig(tomlFilePath)
+	unit, err := detectRuntime()
 	if err != nil {
-		log.Fatalf("Error reading TOML file: %v", err)
+		log.Fatalf("failed to autodetect container runtime: %v", err)
 	}
 
-	updatedTomlMap := updateNRIPlugin(tomlMap)
+	// Edit containerd config only if containerd is detected to be the runtime.
+	if unit == containerdUnit {
+		tomlMap, err := readConfig(containerdConfigFile)
+		if err != nil {
+			log.Fatalf("Error reading TOML file: %v", err)
+		}
 
-	err = writeConfig(tomlFilePath, updatedTomlMap)
+		updatedTomlMap := updateContainerdConfig(tomlMap)
+
+		err = writeToContainerdConfig(containerdConfigFile, updatedTomlMap)
+		if err != nil {
+			log.Fatalf("failed to write updated config into a file %q:, %v", containerdConfigFile, err)
+		}
+	}
+
+	err = writeToCrioConfig()
```

Comment:
Isn't this now unconditionally always trying to update the CRI-O configuration ?